src/common/buffer.hpp

```diff
@@ -284,104 +284,85 @@ class RxCounters {
+            delete[] task_count[i];
     }
 
-private:
-    std::array<size_t, TASK_BUFFER_FRAME_NUM> symbol_count;
-};
-
-class ZF_stats : public Frame_stats {
-public:
-    size_t coded_frame;
-    size_t& max_task_count;
-    ZF_stats(void)
-        : max_task_count(max_symbol_count)
-    {
-    }
-    void init(int max_tasks)
+    /**
+     * Checks whether or not the symbol is the last symbol for a given frame while simultaneously incrementing the symbol count.
+     * If the symbol is the last symbol, the count is reset to 0.
+     * @arg frame id: the frame id to check
+     **/
+    bool last_symbol(int frame_id)
     {
-        Frame_stats::init(max_tasks);
-        coded_frame = SIZE_MAX;
+        const size_t frame_slot = frame_id % TASK_BUFFER_FRAME_NUM;
+        if (++symbol_count[frame_slot] == max_symbol_count) {
+            symbol_count[frame_slot] = 0;
+            return true;
+        }
+        return false;
     }
-};
-
-class Data_stats : public Frame_stats {
-public:
-    size_t max_task_count;
 
-    void init(int _max_task_count, int max_symbols, int max_data_symbol)
-    {
-        Frame_stats::init(max_symbols);
-        for (size_t i = 0; i < TASK_BUFFER_FRAME_NUM; i++)
-            task_count[i] = new size_t[max_data_symbol]();
-        max_task_count = _max_task_count;
-    }
-    void fini()
-    {
-        for (size_t i = 0; i < TASK_BUFFER_FRAME_NUM; i++)
-            delete[] task_count[i];
-    }
-    bool last_task(size_t frame_id, size_t data_symbol_id)
+    /**
+     * Checks whether or not the task is the last task for a given frame and symbol while simultaneously incrementing the task count.
+     * If the task is the last symbol, the count is reset to 0.
+     * @arg frame_id: the frame id to check
+     * @arg symbol_id: the symbol id to check
+     **/
+    bool last_task(size_t frame_id, size_t symbol_id)
     {
         const size_t frame_slot = frame_id % TASK_BUFFER_FRAME_NUM;
-        if (++task_count[frame_slot][data_symbol_id] == max_task_count) {
-            task_count[frame_slot][data_symbol_id] = 0;
+        if (++task_count[frame_slot][symbol_id] == max_task_count) {
+            task_count[frame_slot][symbol_id] = 0;
             return true;
         }
         return false;
     }
 
+    size_t get_symbol_count(size_t frame_id)
+    {
+        return symbol_count[frame_id % TASK_BUFFER_FRAME_NUM];
+    }
+
     size_t get_task_count(size_t frame_id, size_t symbol_id)
     {
         return task_count[frame_id % TASK_BUFFER_FRAME_NUM][symbol_id];
     }
 
 private:
-    size_t* task_count[TASK_BUFFER_FRAME_NUM];
-};
-
-class FFT_stats : public Data_stats {
-public:
-    size_t max_symbol_data_count;
-    std::array<size_t, TASK_BUFFER_FRAME_NUM> symbol_rc_count;
-    size_t max_symbol_rc_count;
+    //array of tasks, per symbol, per frame. This is a moving window over frames.
+    std::array<size_t*, TASK_BUFFER_FRAME_NUM> task_count;
```

Comment:
This can just probably be a 2D array with dimensions `kFrameWnd` and `kMaxSymbols`